querydb/src/main/scala/io/joern/scanners/php/TwigTemplateInjection.scala

```diff
@@ -0,0 +1,122 @@
+package io.joern.scanners.php
+
+import io.joern.console.*
+import io.joern.dataflowengineoss.language.*
+import io.joern.dataflowengineoss.queryengine.EngineContext
+import io.joern.macros.QueryMacros.*
+import io.joern.scanners.*
+import io.shiftleft.codepropertygraph.generated.Operators
+import io.shiftleft.semanticcpg.language.*
+
+object TwigTemplateInjection extends QueryBundle {
+
+  implicit val resolver: ICallResolver = NoResolve
+
+  @q
+  def TwigTemplateInjection()(implicit context: EngineContext): Query =
+    Query.make(
+      name = "twig-template-injection",
+      author = Crew.SJ1iu,
+      title = "Twig-Template-Injection: A parameter controlled by the user is rendered within a Twig template.",
+      description = """
+          |An attacker controlled parameter is used in an twig template.
+          |
+          |This doesn't necessarily indicate a Twig template injection, but if the input is not sanitized and the escape settings are disabled in the application, it could potentially lead to a template injection vulnerability.
+          |""".stripMargin,
+      score = 5,
+      withStrRep({ cpg =>
+
+        def source =
+          cpg.call.name(Operators.assignment).argument.code("(?i).*request.*").l
+
+        def sink =
+          cpg.call.name("createTemplate").l.filter(call => call.methodFullName.toLowerCase.contains("twig")).argument
```

Comment:
the `filter()` here can be more idiomatically expressed as `.methodFullName("(?i).*twig.*")`

Reply:
Hi maltek, thanks for your comments. I've updated in https://github.com/joernio/joern/pull/5148/commits/7471b8a27fc50c2f1ce034654841befc698685f5 as you recommended, however, there is one test failed at `io.joern.rubysrc2cpg.io.RubySrc2CpgHTTPServerTests` after the change. My code has nothing to do with `rubysrc2cpg` module, this error happened once before when I added a new rule for detecting a Java vulnerability, and then it suddenly disappeared without any further changes.. Could you please take a look at it?

Reply:
- you've also deleted the `.argument` at the end. Could you add that back please? Even if it seems to work, it's not really correct: in a dataflow query a call stands for the return value of that call. And for a template we're not interested in the output of template expansion, but only in the input that's used as a template. So we want to use the argument of the call as the sink, not its return value. (Sorry if I wasn't clear about what to change.)
- Github shows that all checks have passed, so the test seems to work in our CI environment. I can't say much about why it'd fail on your machine.